src/test/java/uk/gov/companieshouse/officerfiling/api/controller/DirectorsControllerImplIT.java

```diff
@@ -79,34 +88,53 @@ class DirectorsControllerImplIT {
     OfficerFiling officerFiling;
     @Mock
     OfficerServiceException serviceException;
+    @MockBean
+    private ApiClientService apiClientService;
+    @Mock
+    private ApiClient apiClientMock;
+    @Mock
+    private TransactionsResourceHandler transactionResourceHandlerMock;
+    @Mock
+    private TransactionsGet transactionGetMock;
+    @Mock
+    private ApiResponse<Transaction> apiResponse;
     private HttpHeaders httpHeaders;
     @Autowired
     private MockMvc mockMvc;
 
     @BeforeEach
-    void setUp() {
+    void setUp() throws IOException, URIValidationException {
         httpHeaders = new HttpHeaders();
         httpHeaders.add("ERIC-Access-Token", PASSTHROUGH_HEADER);
         httpHeaders.add("ERIC-Identity", USER);
         httpHeaders.add("ERIC-Identity-Type", KEY);
         httpHeaders.add("ERIC-Authorised-Key-Roles", KEY_ROLE);
-        httpHeaders.add("ERIC-Authorised-Token-Permissions", "company_officers=readprotected,delete,create,update");
+        httpHeaders.add("ERIC-Authorised-Token-Permissions", "company_number="+COMPANY_NUMBER+" company_officers=readprotected,delete,create,update");
+
         var offData = new OfficerFilingData(
                 "etag",
                 null,
                 resignedOn);
+
+        transaction = new Transaction();
+        transaction.setCompanyNumber(COMPANY_NUMBER);
+        transaction.setId(TRANS_ID);
+        transaction.setStatus(TransactionStatus.OPEN);
+
         when(transactionInterceptor.preHandle(any(), any(), any())).thenReturn(true);
         when(openTransactionInterceptor.preHandle(any(), any(), any())).thenReturn(true);
-
-        when(transaction.getCompanyNumber()).thenReturn(COMPANY_NUMBER);
-        when(transaction.getId()).thenReturn(TRANS_ID);
-        when(transaction.getStatus()).thenReturn(TransactionStatus.OPEN);
         when(officerFilingService.get(SUBMISSION_ID, TRANS_ID)).thenReturn(officerFilingOptional);
         when(officerFilingOptional.isPresent()).thenReturn(true);
         when(officerFilingOptional.get()).thenReturn(officerFiling);
         when(officerFiling.getData()).thenReturn(offData);
```

Comment:
Tech debt:
Replace mock `Optional` with explicit stub value:
`when(officerFilingService.get(SUBMISSION_ID, TRANS_ID)).thenReturn(Optional.of(offData));`